### PR TITLE
Fail fast

### PR DESCRIPTION
**Pull request by kolsa**

### CODE REVIEW

**Comment by majst01** [comment]
requires another change in metal-hammer because the renaming of WaitResponse.

Is this a fix for the spurious missed immediate allocations ?

**Comment by kolsa** [comment]
Yes, this could fix that issue.
Also, see [this PR](https://github.com/metal-stack/metal-hammer/pull/24/files)

File: cmd/metal-api/internal/grpc/waitServer.go [hunk 1]
```diff
@@ -159,14 +162,30 @@ func (s *WaitServer) Wait(req *v1.WaitRequest, srv v1.Wait_WaitServer) error {
...
 				return err
 			}
 		}
 	}
 }
 
+// https://github.com/grpc/grpc-go/issues/1229#issuecomment-302755717
+func sendKeepPatientResponse(srv v1.Wait_WaitServer) error {
+	errChan := make(chan error, 1)
+	ctx := srv.Context()
```
Comment (Gerrit91) [inline cmd/metal-api/internal/grpc/waitServer.go:176]: Just for me to know... Under which circumstances does this context receive a `Done()`?

Comment (kolsa) [reply cmd/metal-api/internal/grpc/waitServer.go:176]: Whenever the stream to the client completes or breaks

Comment (Gerrit91) [reply cmd/metal-api/internal/grpc/waitServer.go:176]: Okay, then let's do it.

**Comment by majst01** [comment]
But with this change we also need a new metal-hammer as i said, changing proto definitions in place does not retain compatibility !

**Pull request merged**